### PR TITLE
Fix problems with autobuild setting and affected APIs

### PR DESCRIPTION
**Pull request by stalleyj**
This PR fixes https://github.com/eclipse/codewind/issues/54 so that a project with autobuild set to false will behave as expected. 

The solution was to always call createProject on Turbine even when autobuild is false so that Turbine will set up project metadata and log directories but then only create a build operation when the flag is set to true.  This fixes problems with the capabilities/logs/build apis not working on projects that hadn't been built. problems were particularly encountered when restarting Codewind with a project having autobuild false. I have a clean run of the portal API tests and 4 failures on the Turbine unit tests but these also occurred on master branch.

### CODE REVIEW

File: src/pfe/portal/modules/User.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ module.exports = class User {
       for (let i = 0; i < projectArray.length; i++) {
         let project = projectArray[i];
         // Don't request build if autoBuild is false
-        if (project.isOpen() && project.autoBuild) {
+        if (project.isOpen()) {
```
Comment (rwalle61) [inline src/pfe/portal/modules/User.js]: As far as I can tell, this change means that even if the user has `autoBuild` turned off, the projects will `buildAndRun` when PFE starts. Is that intended?

**Comment by yangcao77** [comment]
This pr does not fix the issue. 
Turbine would call `update` for a manual build. If the project did not build on first time calling `projectCreate`. It would fail.

Portal should call `createProject` whenever the project requires a first time build. Simply set the autobuild flag to false won't work.

**Comment by rajivnathan** [comment]
Discussed with Elson, Toby and Julie this morning. This is the summary: https://github.com/eclipse/codewind/issues/54#issuecomment-527507005

**Comment by tobespc** [comment]
closing as this issue has had a rethink on design

**Pull request closed without merging**